crates/consensus/src/transaction/envelope.rs

```diff
@@ -629,12 +629,173 @@ mod serde_from {
     }
 }
 
+/// Bincode-compatible serde implementation for OpTxEnvelope.
+#[cfg(all(feature = "serde", feature = "serde-bincode-compat"))]
+pub mod serde_bincode_compat {
+    use crate::serde_bincode_compat::TxDeposit;
+    use alloy_consensus::{
+        transaction::serde_bincode_compat::{TxEip1559, TxEip2930, TxEip7702, TxLegacy},
+        Sealed, Signed,
+    };
+    use alloy_primitives::{PrimitiveSignature as Signature, B256};
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+    use serde_with::{DeserializeAs, SerializeAs};
+
+    /// Bincode-compatible representation of an OpTxEnvelope.
+    #[derive(Debug, Serialize, Deserialize)]
+    pub enum OpTxEnvelope<'a> {
+        /// Legacy variant.
+        Legacy {
+            /// Transaction signature.
+            signature: Signature,
+            /// Borrowed legacy transaction data.
+            #[serde(borrow)]
+            transaction: TxLegacy<'a>,
+        },
+        /// EIP-2930 variant.
+        Eip2930 {
+            /// Transaction signature.
+            signature: Signature,
+            /// Borrowed EIP-2930 transaction data.
+            #[serde(borrow)]
+            transaction: TxEip2930<'a>,
+        },
+        /// EIP-1559 variant.
+        Eip1559 {
+            /// Transaction signature.
+            signature: Signature,
+            /// Borrowed EIP-1559 transaction data.
+            #[serde(borrow)]
```

Comment:
ah nice, this way we dont neven need a cow, never thought about this and now I feel stupid :)

Reply:
Super cool, right!